### PR TITLE
feat(hooks): add git poller

### PR DESCRIPTION
**Pull request by bnjjj**

### CODE REVIEW

File: engine/api/hook.go [hunk 1]
```diff
@@ -386,3 +388,89 @@ func processHook(DBFunc func() *gorp.DbMap, store cache.Store, h hook.ReceivedHo
...
+		//get the client for the repositories manager
+		vcsServer := repositoriesmanager.GetProjectVCSServer(proj, vcsServerParam)
+		client, errR := repositoriesmanager.AuthorizedClient(api.mustDB(), api.Cache, vcsServer)
+		if errR != nil {
+			return sdk.WrapError(errR, "getHookPollingVCSEvents> Unable to get client for %s %s", proj.Key, vcsServerParam)
+		}
+
+		//Check if the polling if disabled
+		if info, err := repositoriesmanager.GetPollingInfos(client); err != nil {
+			return err
```
Comment (yesnault) [inline engine/api/hook.go]: wrap?

File: engine/api/hook.go [hunk 2]
```diff
@@ -386,3 +388,89 @@ func processHook(DBFunc func() *gorp.DbMap, store cache.Store, h hook.ReceivedHo
...
+		pullRequestEvents, err := client.PullRequestEvents(h.Config["repoFullName"].Value, events)
+		if err != nil {
+			return sdk.WrapError(err, "getHookPollingVCSEvent> ")
+		}
+
+		repoEvents := sdk.RepositoryEvents{}
+		for _, pushEvent := range pushEvents {
+			exist, errB := workflow.BuildExist(api.mustDB(), h.Config["project"].Value, workflowID, pushEvent.Commit.Hash)
+			if errB != nil {
+				return errB
```
Comment (yesnault) [inline engine/api/hook.go]: wrap?

File: engine/api/api_routes.go [hunk 3]
```diff
@@ -64,6 +64,7 @@ func (api *API) InitRouter() {
 
 	// Hooks
 	r.Handle("/hook", r.POST(api.receiveHookHandler, Auth(false) /* Public handler called by third parties */))
+	r.Handle("/hook/{uid}/workflow/{workflowID}/vcsevent/{vcsServer}", r.GET(api.getHookPollingVCSEvents))
```
Comment (fsamin) [inline engine/api/api_routes.go]: uuid ?

File: engine/api/hook.go [hunk 4]
```diff
@@ -386,3 +388,89 @@ func processHook(DBFunc func() *gorp.DbMap, store cache.Store, h hook.ReceivedHo
...
+			}
+			if !exist {
+				repoEvents.PushEvents = append(repoEvents.PushEvents, pushEvent)
+			}
+		}
+
+		for _, pullRequestEvent := range pullRequestEvents {
+			exist, errB := workflow.BuildExist(api.mustDB(), h.Config["project"].Value, workflowID, pullRequestEvent.Head.Commit.Hash)
+			if errB != nil {
+				return errB
```
Comment (yesnault) [inline engine/api/hook.go]: wrap?

File: engine/api/api_routes.go [hunk 5]
```diff
@@ -368,7 +369,7 @@ func (api *API) InitRouter() {
 	r.Handle("/worker/model/capability/type", r.GET(api.getRequirementTypesHandler))
 
 	// Workflows
-	r.Handle("/workflow/hook", r.GET(api.getWorkflowHooksHandler, NeedService()))
+	r.Handle("/workflow/hook", r.GET(api.getWorkflowHooksHandler))
```
Comment (fsamin) [inline engine/api/api_routes.go]: why remove this ?

File: engine/api/workflow/dao_node_run.go [hunk 6]
```diff
@@ -518,3 +518,19 @@ func updateNodeRunCommits(db gorp.SqlExecutor, id int64, commits []sdk.VCSCommit
 	}
 	return nil
 }
+
+// BuildExist Check if build exist or not
+func BuildExist(db gorp.SqlExecutor, projectKey string, workflowID int64, hash string) (bool, error) {
+	query := `
+	SELECT COUNT(*)
```
Comment (fsamin) [inline engine/api/workflow/dao_node_run.go]: count(1)

File: engine/hooks/tasks.go [hunk 7]
```diff
@@ -199,25 +207,39 @@ func (s *Service) prepareNextScheduledTaskExecution(t *sdk.Task) error {
...
+
+		//Compute a new date
+		t0 := time.Now().In(loc)
+		nextSchedule = cronExpr.Next(t0)
 
-	//Compute a new date
-	t0 := time.Now().In(loc)
-	t1 := cronExpr.Next(t0)
+	case TypeRepoPoller:
+		nextSchedule = time.Now().Add(time.Minute * 2)
```
Comment (fsamin) [inline engine/hooks/tasks.go]: 1 minute.
Could you add a comment to explain all the thing

File: engine/api/workflow/dao_node_run.go [hunk 8]
```diff
@@ -518,3 +518,19 @@ func updateNodeRunCommits(db gorp.SqlExecutor, id int64, commits []sdk.VCSCommit
 	}
 	return nil
 }
+
+// BuildExist Check if build exist or not
+func BuildExist(db gorp.SqlExecutor, projectKey string, workflowID int64, hash string) (bool, error) {
```
Comment (fsamin) [inline engine/api/workflow/dao_node_run.go]: s/Build/Run

**Pull request merged**